src/frontend/src/app/scheduled-process/scheduled-process.component.ts

```diff
@@ -7,48 +7,139 @@ import { SchedulesService } from '../services/index';
+  public scheduleProcess = []
+  public scheduleType = []
+  public tasksData = []
+
+  // To handle field validtion on UI
+  public invalidRepeat: boolean = false
+  public invalidTime: boolean = false
+
+  // Default selected schedule type is STARTUP = 1
+  public selected_schedule_type: Number = 1;
 
   constructor(private schedulesService: SchedulesService) { }
 
   ngOnInit() {
-    this.getSchedules();
-    this.getSchedulesProcesses();
-    this.getLatestTasks();
+    this.getScheduleType()
+    this.getSchedules()
+    this.getSchedulesProcesses()
+    this.getLatestTasks()
+  }
+
+  public getScheduleType(): void {
+    this.schedulesService.getScheduleType().
+      subscribe(
+      data => {
+        this.scheduleType = data.schedule_type
+        console.log(this.scheduleType)
+      },
+      error => { console.log("error", error) })
   }
 
-  public getSchedules():void {
-    this.schedulesData = [];
+  public getSchedules(): void {
+    this.scheduleData = [];
     this.schedulesService.getSchedule().
       subscribe(
       data => {
-        this.schedulesData = data.schedules;
-        console.log("This is the getSchedules ",  data.schedules);
+        this.scheduleData = data.schedules
+        console.log("This is the getSchedule ", data.schedules)
       },
       error => { console.log("error", error) });
   }
 
-  public getSchedulesProcesses():void {
-    this.schedulesProcessData = [];
+  public getSchedulesProcesses(): void {
+    this.scheduleProcess = [];
     this.schedulesService.getScheduledProcess().
       subscribe(
       data => {
-        this.schedulesProcessData = data.processes;
-        console.log("This is the getSchedulesProcesses ",  this.schedulesProcessData);
+        this.scheduleProcess = data.processes
+        console.log("This is the getScheduleProcess ", this.scheduleProcess)
       },
       error => { console.log("error", error) });
   }
 
-  public getLatestTasks():void {
+  public getLatestTasks(): void {
     this.tasksData = [];
     this.schedulesService.getLatestTask().
       subscribe(
       data => {
         this.tasksData = data.tasks;
-        console.log("This is the tasks ",  data.tasks);
+        console.log("Tasks data", data.tasks)
       },
-      error => { console.log("error", error) });
+      error => { console.log("error", error) })
+  }
+
+  /**
+   *  To set schedule type key globally for required field handling on UI
+   * @param value  
+   */
+  public setScheduleTypeKey(value) {
+    this.selected_schedule_type = value;
+  }
+
+  public createSchedule() {
+    let schedule_name = <HTMLInputElement>document.getElementById("name")
+    let schedule_process = <HTMLSelectElement>document.getElementById("process")
+    let schedule_type = <HTMLSelectElement>document.getElementById("type")
+    let repeat_interval = <HTMLInputElement>document.getElementById("repeat")
+    let exclusive_state = <HTMLInputElement>document.getElementById("exclusive")
+
```

Comment:
```
let schedule_name = <HTMLInputElement>document.getElementById("name").value
let schedule_process = <HTMLSelectElement>document.getElementById("process").value
let schedule_type = <HTMLSelectElement>document.getElementById("type").value
let exclusive_state = <HTMLInputElement>document.getElementById("exclusive").checked
```

and then

let repeat_interval_fld = <HTMLInputElement>document.getElementById("repeat")
then use this to calculate var repeat_time

Also let's have var scheduled_time and var day here with default null value? and then in if `(this.selected_schedule_type == 2) {` check scheduled_time validity and assign this and the day from field